etc/init.d/contextBroker.centos

```diff
@@ -94,11 +96,56 @@ broker_start()
         # logs are always managed by CB and written to file inside directory BROKER_LOG_DIR (see /etc/sysconfig/contextBroker) and rotated by logrotate
         BROKER_STDERR_LOG=${BROKER_LOG_DIR}/${NAME}-error.log
         su $BROKER_USER -c "${EXECUTABLE} ${BROKER_OPTS} 1>/dev/null 2>${BROKER_STDERR_LOG}"
+        sleep 2 # wait some time to leave contextbroker start
+        PID=$(cat ${BROKER_PID_FILE})
+        if [[ -z "${PID}" ]]; then
+            printf "%s" "pidfile not found"
+            printf "%s\n" "$(failure)"
+            exit 1
+        else
+            printf "%s\n" "$(success)"
+        fi
+    else
+        printf "%s\n" "Fail - missing ${EXECUTABLE} executable"
+        exit 1
+    fi
+
+}
+
+rotate_cores ()
+{
+    cd /var/cb_cores
+    STR_DATE=$(date +"%Y%m%d_%H%M%S")
+    if [[ $(ls core* 2>/dev/null | wc -l) -gt 0 ]]; then 
+        cp ${BROKER_LOG_DIR}/contextBroker.log . 
+        tar jcf CB_core_${STR_DATE}.tar.bz2 core* contextBroker.log
+        rm -f core* contextBroker.log
+    fi
+
+    TOTAL=$(ls CB_core_* 2>/dev/null | wc -l)
+    KEEP_CORE=8
+    if [[ ${TOTAL} -gt ${KEEP_CORE} ]]; then
+        rm -f $(ls -tr 2>/dev/null |  head -$((${TOTAL} - ${KEEP_CORE})))
+    fi
+
+}
+
+broker_coredump_start()
+{
+    printf "%-50s" "Starting ${NAME}..."
+    if [[ -x ${EXECUTABLE} ]]; then
+        mkdir -p /var/cb_cores &> /dev/null
+        cd /var/cb_cores
+        rotate_cores &
+        ulimit -c unlimited
+        ulimit -n {{ orion_config.orion_root_max_open_files | mandatory }}
+        ${EXECUTABLE} ${BROKER_OPTS} -fg &> /dev/null &
```

Comment:
Which users starts Orion in coredump case? How do you ensure that user is root (as *"When this var is 'true' 
the process is started with 'root' user"* says).

Reply:
You are right I cannot ensure the `root`  user but probably the script fails in somewhere before the contextBroker starts. So, if the contextBroker starts it will be as root user with very high probability 😄

Reply:
Another option would be to override the BROKER_USER option in the case of coredump and use root to do the execution, i.e.:

```
su root -c "${EXECUTABLE} ${BROKER_OPTS} -fg &> /dev/null &"
```
What I'm not sure if it that is going to work... not sure of su + -fg + & work together...

Reply:
After talking with @vgarciag , the best solution is keep the current way but checking BROKER_USER is 'root' previously (otherwise exit with an error message).

Reply:
Fixed in 48afdc3740503daca08277b7df9ff9e3d69f6e57